### DISCUSSION TITLE
Практическое использование Source.getArtists

### DISCUSSION THREAD

**Discussion by Bedmit** [Chimildic/goofy#22] [Помощь с алгоритмом] [answered]
Не могу понять, как использовать функцию **getArtists**
Например для задачи удаления из истории прослушивания треков любимых исполнителей, использую скрипт:
```
  let History = RecentTracks.get();
  let Minions = Source.getArtistsTracks({
    artist: {
    followed_include: true,},
  });
  Filter.removeArtists(History, Minions);
```
Но мне кажется вместо получения массива всех треков любимых исполнителей, достаточно массива только этих исполнителей полученных функцией   **getArtists**. Но в таком случае фильтр **Filter.removeArtists** выдает ошибку.

**Comment by vlalog** [comment]
`Source.getArtists()` возвращает массив исполнителей, а не треков.
Этот массив можно использовать в `Filter.removeArtists()`, если не ошибаюсь

**Comment by vlalog** [comment]
Скорее всего вам нужен [этот метод](https://chimildic.github.io/goofy/#/func?id=getartiststracks)

**Reply by Bedmit** [reply]
Так именно им в настоящее время и пользуюсь (скрипт показан в первом сообщении).
В том то и дело, что все треки мне не нужны, если достаточно только списка исполнителей. Но **Filter.removeArtists()** с массивом **Source.getArtists()** не работает. 
Поэтому не очень понимаю, как вообще эту функцию использовать. И можно ли упростить мой скрипт.

**Comment by Chimildic** [comment, marked as answer]
`removeArtists` умеет брать только `id` исполнителя **из трека**. `getArtists` дает только исполнителей, без треков. Поэтому ошибка.
 
Раньше в потоках были только треки, все затачивалось под них. Теперь появились альбомы и исполнители. И некоторые функции вызывают вопросы. Хотя на уровне описания аргументов все правильно. Если посмотреть в [removeArtists](https://chimildic.github.io/goofy/#/func?id=removeartists), второй аргумент это массив треков, а не исполнителей.

Внесу правки, чтобы `removeArtists` мог удалять и по исполнителю без трека.

`getArtists` можно использовать в связке с получением сначала альбомов, а потом треков из них. Подобная реализация есть в [getArtistsTracks](https://chimildic.github.io/goofy/#/func?id=getartiststracks). Но на случай **своих модификаций**, были открыты `getArtists` и прочие. 
Или в других функциях, где явно указан массив исполнителей. Например, [getRelatedArtists](https://chimildic.github.io/goofy/#/func?id=getrelatedartists)

**Reply by Chimildic** [reply]
> Внесу правки, чтобы removeArtists мог удалять и по исполнителю без трека.

Добавил правку в версию 1.3.3
